tests/integration/cmd_debug_test.go

```diff
@@ -0,0 +1,66 @@
+package integration
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo debug command tests", func() {
+	var project string
+	var context string
+
+	// Setup up state for each test spec
+	// create new project (not set as active) and new context directory for each test spec
+	// This is before every spec (It)
+	BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		SetDefaultConsistentlyDuration(30 * time.Second)
+		context = helper.CreateNewContext()
+		project = helper.CreateRandProject()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	AfterEach(func() {
+		helper.DeleteProject(project)
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("odo debug on a nodejs component", func() {
+		It("should expect a ws connection when tried to connect on different debug port locally and remotely", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project, "--context", context)
+			helper.CmdShouldPass("odo", "config", "set", "--force", "DebugPort", "9292", "--context", context)
+			dbgPort := helper.GetConfigValueWithContext("DebugPort", context)
+			Expect(dbgPort).To(Equal("9292"))
```

Comment:
This should be validated in dc too after the push.

Reply:
Actually, if it works then it should be present in dc.

Reply:
Explicit is better than implicit. :wink:

Reply:
will resolve this in the followup PR

Reply:
> Actually, if it works then it should be present in dc.

In case it fails then how you confirm the port is added in dc but failing due to some other reason. Moreover this pr is  responsible to add the port in dc, so it should be validated just to avoid regression in future.